src/codegen/ast_interpreter.cpp

```diff
@@ -285,32 +285,35 @@ ASTInterpreter::ASTInterpreter(CLFunction* clfunc)
     assert(scope_info);
 }
 
-void ASTInterpreter::initArguments(int nargs, BoxedClosure* _closure, BoxedGenerator* _generator, Box* arg1, Box* arg2,
-                                   Box* arg3, Box** args) {
+void ASTInterpreter::initArguments(BoxedClosure* _closure, BoxedGenerator* _generator, Box* arg1, Box* arg2, Box* arg3,
+                                   Box** args) {
+    RootedBox arg1_root(arg1);
+    RootedBox arg2_root(arg2);
+    RootedBox arg3_root(arg3);
```

Comment:
have you made much progress on figuring out why these are needed?  I wouldn't mind punting on that issue for now, but we should have some sort of comment explaining why we have these RootedBoxes here.